src/blog/2024/10/how-to-build-automate-devops-pipelines-node-red-deployments.md

```diff
@@ -0,0 +1,95 @@
+8. Choose an instance, device, or device group for the stage.
+9. Next, configure which action should be performed when this stage is pushed to the next:
+   - **Create New Snapshot:** Generates a new [Snapshot](/docs/user/high-availability/) using the current flows and settings.
+   - **Use Latest Instance Snapshot:** Uses the most recent existing snapshot of the instance. The deployment will fail if no snapshot exists.
+   - **Prompt to Select Snapshot:** Prompts at deploy time to select which snapshot from the source stage should be copied to the next stage.
+10. Check the option **Deploy to Devices** if you want changes to be deployed to all devices connected to this stage’s instance when the stage is deployed.
+
+![Screenshot showing the stage configuration form with options to select instance types and configure actions.](./images/form-to-create-configure-stages.png){data-zoomable}
+_Configure each stage by selecting an instance, device, or device group, and define the deployment actions_
+
+Once you’ve created your initial stage, you can add more stages by following the same process. This flexibility allows you to tailor your DevOps pipeline to meet the specific needs of your Node-RED deployment.
+
+For example, in development, you might have a Node-RED instance in the cloud to build your application. During staging, you could test the setup with a single device. Finally, in production, you can deploy the tested application to thousands of devices in a device group, saving time and ensuring smooth deployment at scale.
+
+![DevOps pipelines animation](./images/devops-pipeline.gif){data-zoomable}
+*Image: DevOps animation showing application deployment to production securely.*
+
+## Running a Pipeline Stage
+
+Once your pipeline is set up, you are ready to run it and deploy your changes across each stage. Here's how to do it:
+
+![Screenshot of the 'Run Pipeline' button in FlowFuse, allowing users to trigger the deployment process.](./images/button-to-run-pipeline.png){data-zoomable}
+_Click the 'Run Pipeline' button to initiate the a deployment process across for the selected given stage_
+
+1. Click the run pipeline stage option of first stage. For each stage, this button will be available, except for the last one. Simply click it to start the deployment process.
+2. Once initiated, the deployment will automatically progress from the current stage to the next stage on the right. For example, if you have three stages—Development, Staging, and Production—clicking the "Run Pipeline" button for the Development stage will advance the deployment to the Staging stage. To proceed from Staging to Production, you will need to click the "Run" button for the Staging stage, as each pair of adjacent stages operates independently.
+
+This action will create a new snapshot in the current stage and copy all settings, environment variables, and flows to the next stage, but it will not overwrite any existing environment variable keys in the target stage.
```

Comment:
This section is now far too wordy and confusing.

`1` starts by talking about the "first stage" then goes on to mention "each stage"
`2` continues on as if you never mentioned "each stage" in `1`)
the example added to `2` is very long and detracts from the simple steps

the following sentence state "This action will create a new snapshot in the current stage" but i am lost as to which stage you are referring to.